### ISSUE TITLE
[Validator] Validator ignores $constraints argument

### ISSUE THREAD

**Issue by BoShurik** [symfony/symfony#28257]
**Symfony version(s) affected**: 4.1.3

**Description**  
When passing constraint to `validate` method I've got violations from annotation constrains.

**How to reproduce**  
https://github.com/BoShurik/symfony-validator-issue

```php
class Foo
{
    /**
     * @var Bar[]
     *
     * @Assert\Valid()
     */
    public $bars;

    public function __construct()
    {
        $this->bars = [
            new Bar(),
            new Bar(),
        ];
    }
}
```

```php
$violations = $this->validator->validate($model->bars, new Count([
    'min' => 3,
    'max' => 3,
]));
/** @var ConstraintViolationInterface $violation */
foreach ($violations as $violation) {
    dump(sprintf('%s: %s', $violation->getPropertyPath(), $violation->getMessage()));
}

/*
    Got:
    ": This collection should contain exactly 3 elements."
    "[0].name: This value should not be blank." <- not expected
    "[1].name: This value should not be blank." <- not expected
 */
```

**Comment by xabbuh** [comment]
Status: Reviewed

**Comment by xabbuh** [comment]
probably related to #27090

**Comment by modiamir** [comment]
@xabbuh Does it need to be fixed?

**Comment by xabbuh** [comment]
I can no longer reproduce it. Probably fixed by #29800.
